django-stubs/db/migrations/operations/special.pyi

```diff
@@ -17,14 +17,14 @@ class SeparateDatabaseAndState(Operation):
 
 class RunSQL(Operation):
     noop: Literal[""]
-    sql: str | _ListOrTuple[str | tuple[str, dict[str, Any] | _ListOrTuple[str] | None]]
-    reverse_sql: str | None | _ListOrTuple[str | tuple[str, dict[str, Any] | _ListOrTuple[str] | None]]
+    sql: str | _ListOrTuple[str | tuple[str, dict[str, Any] | _ListOrTuple[Any] | None]]
+    reverse_sql: str | None | _ListOrTuple[str | tuple[str, dict[str, Any] | _ListOrTuple[Any] | None]]
     state_operations: Sequence[Operation]
     hints: Mapping[str, Any]
     def __init__(
         self,
-        sql: str | _ListOrTuple[str | tuple[str, dict[str, Any] | _ListOrTuple[str] | None]],
-        reverse_sql: str | None | _ListOrTuple[str | tuple[str, dict[str, Any] | _ListOrTuple[str] | None]] = ...,
+        sql: str | _ListOrTuple[str | tuple[str, dict[str, Any] | _ListOrTuple[Any] | None]],
```

Comment:
We now don't need `_ListOrTuple[str | tuple[str, dict[str, Any]` part

Reply:
Are you sure about that ? I feel like every variant is needed here: sql il either:
- `str`: -> `RunSQL(sql="SOME SQL")`
- `_ListOrTuple[str]` -> `RunSQL(sql=("SOME SQLS", "SOME SQLS"))` / `RunSQL(sql=["SOME SQLS", "SOME SQLS"])`
- `_ListOrTuple[tuple[str, dict[str, Any]]` -> `RunSQL(sql=[("SOME SQLS %(VAL)s", {"VAL": "FOO"})]`
- `_ListOrTuple[tuple[str, _ListOrTuple[Any]]` ->`RunSQL(sql=[("SOME SQLS %s, %s", [object(), "ANOTHER PARAM"])])`
- `_ListOrTuple[tuple[str, None]` -> `RunSQL([("INSERT INTO musician (name) VALUES ('Reinhardt');", None)])`

I don't really see how it can be simplified, am I missing something here ?

btw, I'v added another test case and a TypeAlias to reduce duplication

Reply:
We have `_ListOrTuple[Any]`, `Any` matches all possible inner types.

Reply:
This is only part of the union for the second member of the tuple, not for the whole thing
```python
str | 
_ListOrTuple[
    str |
    tuple[
        str, 
        dict[str, Any] | _ListOrTuple[Any] | None
    ]
]
```